File: cracking_the_coding_interview/1.2permutation.java
import java.io.*;
import java.util.*;



class Solution {
  public static void main(String[] args) {
    System.out.println(permutation("abc", "acg"));
  }

  public static String Sort(String str){
    char[] arrStr = str.toCharArray();
    java.util.Arrays.sort(arrStr);
    return new String(arrStr);
  }
  
  public static Boolean permutation(String str1, String str2)
  {
    if (str1.length() != str2.length()) return false;
    
    if (Sort(str1).equals(Sort(str2))) return true;
    
    return false;
  }
}

File: cracking_the_coding_interview/1.2permutation.cs
using System;

// To execute C#, please define "static void Main" on a class
// named Solution.

class Solution
{
    static void Main(string[] args)
    {
        /*for (var i = 0; i < 5; i++)
        {
            Console.WriteLine("Hello, World");
        }*/
        
        Console.WriteLine(permutation("abc", "bac"));
        
        
    }
    
    static bool permutation(string str1, string str2)
    {
        char[] arrStr1 = str1.ToCharArray();
        char[] arrStr2 = str2.ToCharArray();
        
        Array.Sort(arrStr1);
        Array.Sort(arrStr2);
        
        string newStr1 = new string(arrStr1);
        string newStr2 = new string(arrStr2);
        
        if (newStr1.Equals(newStr2)) return true;
        
        return false;
    }
}